### ISSUE TITLE
Certain HTML file crashes Atom (w/ Tree Sitter)

### ISSUE THREAD

**Issue by michaelblyons** [atom/language-html#211]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Opening a particular html file with tree-sitter grammars enabled crashes Atom.

### Steps to Reproduce

1. Open Atom 1.29.0 (safe mode or otherwise).
2. Make sure you have Tree Sitter grammars enabled.
3. Open [this file](https://github.com/deathaxe/Packages/blob/ed96c11069996cf68935b92c257d1ebe2935c694/HTML/syntax_test_html.html).

**Expected behavior:** File opens and syntax highlighting is applied

**Actual behavior:** Atom crashes

**Reproduces how often:** 100%

### Versions

```
[michael ~ (master)]$ atom --version
Atom    : 1.29.0
Electron: 2.0.5
Chrome  : 61.0.3163.100
Node    : 8.9.3
[michael ~ (master)]$ apm --version
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.29.0
python 2.7.15
git 2.18.0
```

macOS 10.13.6

**Comment by Arcanemagus** [comment]
Permalink to the version in question: [syntax_test_html.html](https://github.com/sublimehq/Packages/blob/289782ff2e4cb58de171579c7fc86fe00d280619/HTML/syntax_test_html.html).

I'm unable to replicate this on Atom v1.32.0-nightly2, so it looks like whatever issue was causing the crash has been fixed since Atom v1.29.0.

Thanks for taking the time to report this!

**Comment by michaelblyons** [comment]
@Arcanemagus Thank you for the swift reply. I did not realize I was posting on the day that 1.30 would come out. It turns out, I also lied a little: I'm using a branch of the main Sublime Text syntax definitions that includes [deathaxe:pr/html-fix-1565](https://github.com/deathaxe/Packages/tree/pr/html-fix-1565), which I am now using to test the behavior.

On 1.30, [the file](https://github.com/deathaxe/Packages/blob/ed96c11069996cf68935b92c257d1ebe2935c694/HTML/syntax_test_html.html) usually loads, but highlighting becomes corrupted partway through by [the `<![CDATA[` tests](https://github.com/deathaxe/Packages/blob/ed96c11069996cf68935b92c257d1ebe2935c694/HTML/syntax_test_html.html#L119-L146), and continues to have effects through the rest of the file.

Before identifying the `<![CDATA[` as the culprit, I worked backwards from [the first place I noticed weird highlighting](https://github.com/deathaxe/Packages/blob/ed96c11069996cf68935b92c257d1ebe2935c694/HTML/syntax_test_html.html#L282), specifically: the closing tag is not colored.

``` html
        <tag attr otherattr attr-with-dashes attr_with_underscores attr.with.dot attr.with.dot.value="val"></tag>
```

I figured the trouble might stem from the odd set of attributes, so I deleted them. Usually that alone causes Atom to crash, although sometimes it waits until I try to change the opening tag from `tag` to `a`.

Sorry I don't have the 1.32 nightly.

**Comment by Arcanemagus** [comment]
Ah, it helps when we have the correct file to test things with 😉.

Confirming a crash in `v1.32.0-nightly2` on that file, I'll go ahead and update the reproduction steps for that!

**Comment by maxbrunsfeld** [comment]
Fixed via https://github.com/tree-sitter/tree-sitter-html/pull/3.

**Comment by maxbrunsfeld** [comment]
This fix will go out soon in Atom 1.31.0-beta1.
